twitter_api/main.py

```diff
@@ -1,28 +1,217 @@
+                    INNER JOIN user as u ON t.user_id=u.id 
+                    WHERE t.id = ?'''
+    result_cursor = g.db.execute(sql_command, [tweet_id])
+    result_tuple = result_cursor.fetchone()
+    if not result_tuple:
+        abort(404)
+        
+    tweet_id, user_id, created, content, username = result_tuple
+    uri = url_for('get_tweet', tweet_id = tweet_id) # good note from Santiago
 
-@app.errorhandler(404)
+    profile = url_for('profile', username = username)
+    
+    return_dict = dict(id=tweet_id, content=content,
+                    date=_convert_date_iso_8601(created), 
+                    profile=profile, uri=uri)
+    response = Response(
+        json.dumps(return_dict), status=200, content_type=JSON_MIME_TYPE)
+    return response
+    
+@app.route('/tweet/<tweet_id>', methods = ['DELETE'])
+@json_only
+@auth_only
+def delete_tweet(tweet_id):
+    tweet_user_id = _get_tweet_info(tweet_id)[1]
+    
+    # we know access_token is good, so get the client_user_id
+    client_user_id = _get_user_id_from_token(request)
+    
+    if tweet_user_id != client_user_id:
+        abort(401)
+        
+    # Delete entire row from tweet table
+    sql_command = 'DELETE FROM tweet WHERE id = ?'
+    g.db.execute(sql_command, [tweet_id])
+    g.db.commit()
+    return Response('', 204) # No Content
+    
+@app.route('/tweet', methods = ['POST'])
+@json_only
+@auth_only
+def create_tweet():
+    content = request.get_json()['content']
+    client_user_id = _get_user_id_from_token(request)
+    
+    sql_command = 'INSERT INTO tweet (user_id, content) VALUES (?, ?)'
+    g.db.execute(sql_command, [client_user_id, content])
+    g.db.commit()
+    
+    return Response('Created', 201)
+
+@app.errorhandler(404) # Not found
 def not_found(e):
-    return '', 404
+    return Response('', 404)
 
+@app.errorhandler(401) # Unauthorized
+def not_found(e):
+    return Response('', 401)
 
-@app.errorhandler(401)
+@app.errorhandler(400) # Bad request
 def not_found(e):
-    return '', 401
+    return Response('', 400)
+
+@app.route('/profile/<username>')
+def profile(username):
+
+    sql_command1 = 'SELECT id, username, first_name, last_name, birth_date FROM user WHERE username = ?'
+    user_db_data = g.db.execute(sql_command1, [username])
+    user_data = user_db_data.fetchone()
+    
+    if not user_data:
+        abort(404)
+
+    user_id = user_data[0]
+
+    dict_keys = ("user_id", "username", "first_name", "last_name", 
+            "birth_date")
+    profile_json_data = dict(zip(dict_keys, user_data))
+
+    # add tweets to the json_data dictionary
+    sql_command2 = 'SELECT id, created, content, id FROM tweet WHERE user_id = ?'
+    user_tweets = g.db.execute(sql_command2,[user_id])
+    tweet_list = []
+    for tweet in user_tweets:
+        dict_keys = ("id", "date", "text", "uri")
+        row_dict = dict(zip(dict_keys, tweet))
+        row_dict['uri'] = '/tweet/{}'.format(row_dict['uri'])
```

Comment:
I actually prefer your way of pulling the 'plain' uri (id) from the ```tweet``` table and then formatting it, however you could if you preferred directly include the formatting in the sql query ```sql_command2``` as:
```SQL
SELECT id, created, content, "/tweet/" || id FROM tweet WHERE user_id = ?
``` 
as ```||``` is the string concatenation operator in SQLite.